lib/membrane/core/telemetry.ex

```diff
@@ -166,108 +160,71 @@ defmodule Membrane.Core.Telemetry do
   end
 
   @doc """
-  Reports an arbitrary span of a function consistent with `span/3` format in `:telementry`
+  Reports a span of a compoment callback function in a format consistent with `span/3` in `:telementry`
   """
-  @spec component_span(module(), atom(), (-> {:telemetry, telemetry_callback_metadata()})) ::
-          any()
-  def component_span(component_type, callback, f) do
-    component_type = state_module_to_atom(component_type)
+  @spec span_component_callback(
+          (-> CallbackHandler.callback_return() | no_return()),
+          module(),
+          atom(),
+          Telemetry.callback_span_metadata()
+        ) :: CallbackHandler.callback_return() | no_return()
+  def span_component_callback(f, component_module, callback, meta) do
```

Comment:
```suggestion 
 def apply_callback_within_span(callback_lambda, component_module, callback, meta) do
```
IMO name of this function is little weird

Reply:
It uses span as a verb, but I'll find a better name for it

Reply:
What do you think of "track_callback_handler" it has less ambiguous verb, it still specifies it's for a callback and also adds handler to be more descriptive and consistent with "callback_tracked?/1" function name